Cython/Utility/Builtins.c

```diff
@@ -123,12 +123,16 @@ static PyObject* __Pyx_PyExec3(PyObject* o, PyObject* globals, PyObject* locals)
     {
         // For the limited API we just defer to the actual builtin
         // (after setting up globals and locals) - there's too much we can't do otherwise
-        PyObject *builtins, *exec;
+        PyObject *builtins, *exec, *exec_str;
         builtins = PyEval_GetBuiltins();
         if (!builtins) return NULL;
-        exec = PyDict_GetItemString(builtins, "exec");
+        exec_str = PyUnicode_FromStringAndSize("exec", 4);
+        if (!exec_str) return NULL;
+        exec = PyObject_GetItem(builtins, exec_str);
```

Comment:
Can't we use `PyDict_GetItemStringRef` here and below?

Reply:
I think the reason for this was that `PyDict_GetItem*` don't treat it as an error if the key doesn't exist.

So the existing code did the wrong thing if it doesn't find `exec` in the builtins (although I don't know why that would happen).

So `PyObject_GetItem` is trading the convenience of having the exception generated for a little bit of lost performance.

These are both Limited API so not actually a free-threading issue right now, but it was just something I spotted while searching for `PyDict_GetItem`

Reply:
Got it.